pkg/controller/master/virtualmachine/vm_controller.go

```diff
@@ -524,3 +532,134 @@ func (h *VMController) getPVCStorageClass(pvc *corev1.PersistentVolumeClaim) (*s
 	}
 	return sc, nil
 }
+
+// parse cloud-init network-data from VM
+func (h *VMController) getNetworkData(vm *kubevirtv1.VirtualMachine) (string, error) {
+        for _, vol := range vm.Spec.Template.Spec.Volumes {
+                if vol.CloudInitNoCloud == nil {
+                        continue
+                }
+
+                if vol.CloudInitNoCloud.NetworkDataSecretRef == nil {
+                        continue
+                }
+                secName := vol.CloudInitNoCloud.NetworkDataSecretRef.Name
+                sec, err := h.secretCache.Get(vm.Namespace, secName)
+                if err != nil {
+                        return "", err
+                }
+
+                netDataB64, ok := sec.Data["networkdata"]
+                if !ok {
+                        continue
+                }
+                netData, err := base64.StdEncoding.DecodeString(string(netDataB64))
+                if err != nil {
+                        return "", err
+                }
+                return string(netData), nil
+        }
+        return "", nil
+}
+
+// parseAllStaticIPs returns map[interface]ipCIDR
+func (h *VMController) parseAllStaticIPs(networkData string) map[string]string {
+	m := make(map[string]string)
+	ethernets := gjson.Get(networkData, "ethernets")
+	ethernets.ForEach(func(iface, cfg gjson.Result) bool {
+		if ip := cfg.Get("addresses.0").String(); ip != "" {
+			m[iface.Str] = ip
+		}
+		return true
+	})
+	return m
+}
+
+func (h *VMController) parseMAC(networkData, iface string) string {
+	return gjson.Get(networkData, "ethernets."+iface+".macaddress").String()
+}
+
+func kubeOvnAnnoKeys(nadName, nadNS string) (ipKey, macKey string) {
+	// nadName may contain dots -> replace to avoid ambiguity
+	safeName := strings.ReplaceAll(nadName, ".", "-")
+	return fmt.Sprintf(util.AnnotationPatternKubeOvnIP, safeName, nadNS),
+		fmt.Sprintf(util.AnnotationPatternKubeOvnMAC, safeName, nadNS)
+}
+
+// logicalIfaceName returns the Linux interface name that KubeVirt will create
+// for the n-th multus network: eth1, eth2, … (eth0 is always the primary pod network).
+func logicalIfaceName(index int) string {
+	return fmt.Sprintf("eth%d", index)
```

Comment:
So, how to guarantee the naming is always start from `eth`, if the guest OS has systemd-udev with predictable naming?